kie-server-parent/kie-server-router/kie-server-router-proxy/src/main/java/org/kie/server/router/Configuration.java

```diff
@@ -62,23 +62,17 @@ public Configuration() {
     }
 
     public void addContainerHost(String containerId, String serverUrl) {
-        List<String> hosts = hostsPerContainer.get(containerId);
-        if (hosts == null) {
-            hosts = new ArrayList<>();
-            hostsPerContainer.put(containerId, hosts);
-        }
-        hosts.add(serverUrl);
-
+        List<String> list = hostsPerContainer.computeIfAbsent(containerId, x -> new ArrayList<>());
+        if (!list.contains(serverUrl))
+          list.add(serverUrl);
```

Comment:
The idea behind this is to avoid adding duplicates to the lists of hostsPerContainer. computeIfAbsent returns the list (new or existing) and then we can add only in case the list doesn't contain that element (using Set instead of List could be another option but probably implies reworking other parts).